pkg/api/steve/supportconfigs/generator.go

```diff
@@ -0,0 +1,168 @@
+package supportconfigs
+
+import (
+	"archive/tar"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	v1 "github.com/rancher/rancher/pkg/generated/norman/core/v1"
+	"github.com/rancher/rancher/pkg/types/config"
+	"github.com/sirupsen/logrus"
+	authzv1 "k8s.io/api/authorization/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	authv1 "k8s.io/client-go/kubernetes/typed/authorization/v1"
+)
+
+const (
+	// Endpoint The endpoint that this URL is accessible at - used for routing and for SARs
+	Endpoint            = "/v1/generateSUSERancherSupportConfig"
+	cspAdapterNamespace = "cattle-csp-system"
+	cspAdapterConfigmap = "csp-config"
+	tarContentType      = "application/x-tar"
+	logPrefix           = "support-config-generator"
+)
+
+type GeneratorHandler struct {
+	ConfigMaps           v1.ConfigMapInterface
+	SubjectAccessReviews authv1.SubjectAccessReviewInterface
+	ConfigHandler        supportConfigHandler
+}
+
+func NewGeneratorHandler(scaledContext *config.ScaledContext) GeneratorHandler {
+	return GeneratorHandler{
+		ConfigMaps:           scaledContext.Core.ConfigMaps(cspAdapterNamespace),
+		SubjectAccessReviews: scaledContext.K8sClient.AuthorizationV1().SubjectAccessReviews(),
+		ConfigHandler:        supportConfigHandler{},
+	}
+}
+
+func (h *GeneratorHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	authorized, err := h.authorize(request)
+	if err != nil {
+		_, _ = writer.Write([]byte("forbidden"))
+		writer.WriteHeader(http.StatusForbidden)
```

Comment:
nit: 401, no user was authorized. The 403 on line 54 seems correct.

Reply:
I originally had a setup which distinguished between 401 (can't get creds), 403 (got creds SAR denied) and 500 (I got an error when trying to auth). @cmurphy Pointed out that at this stage similar functionality returns a 403 as a catch-all.

I can revert this to an earlier setup, but I do think that 403 makes sense in this context given that we are protected by an authn route earlier on. I'll bring this up in standup so we can make sure everyone is on the same page.

Reply:
Na, just leave as 403 everywhere. There's also the theory that we should return a 404 everywhere here as to avoid "leaking" a valid endpoint to a person without access, but that's trading security for a poor UX.